<?xml version="1.0"?>
<launch>
  <arg name="traj_file" default="traj_1"/>
  <arg name="gui" default="0"/>
  <arg name="remote_launch" default="0"/>
  <include if="$(arg remote_launch)" file="$(find omniveyor_mobility)/launch/agv.machine"/>
  <include unless="$(arg remote_launch)" file="$(find omniveyor_mobility)/launch/local.machine"/>
  <!-- Trajectory reader-->
  <node machine="AGV" pkg="omniveyor_mobility" type="create_trajectory.py" name="trajectory_writer" output="screen">
    <param name="traj_file" value="$(find omniveyor_mobility)/resources/trajectories/$(arg traj_file).txt"/>
  </node>
  <!-- Visualization -->
  <group if="$(arg gui)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find omniveyor_mobility)/resources/movebase.rviz"/>
  </group>
</launch>
